tsdb/shard.go

```diff
@@ -252,19 +491,22 @@ func (s *Shard) ValidateAggregateFieldsInStatement(measurementName string, stmt
 
 // deleteSeries deletes the buckets and the metadata for the given series keys
 func (s *Shard) deleteSeries(keys []string) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
 	if err := s.db.Update(func(tx *bolt.Tx) error {
 		b := tx.Bucket([]byte("series"))
 		for _, k := range keys {
 			if err := b.Delete([]byte(k)); err != nil {
 				return err
 			}
-			if err := tx.DeleteBucket([]byte(k)); err != nil {
+			if err := tx.DeleteBucket([]byte(k)); err != nil && err != bolt.ErrBucketNotFound {
```

Comment:
This looks like it might be fixing another error that people have reported. Some users have complained of "bucket not found" coming back to them.